File: test/nodejs/TestMySQL.js
console.log("To exit, hold down [Ctrl] and press [C] twice.");

//////////
console.log("Test MySQL");

var mysql = require("mysql");
var pool = mysql.createPool({
  database: 'test',
  user: 'root',
  password: 'seigi@123',
  host: 'localhost',
  port: 3306,
});

pool.getConnection(function(err, client) {
  if (err) {
    console.log(err);
  } else {
    client.query('SELECT * FROM Shippers', function (error, obj, fields) {
      console.log(obj);
    });
  }
});

File: test/python/test_mysql.py
"""MySQL 接続テスト（PyMySQL）。"""
import pymysql

from config import CONNECT_TIMEOUT_SEC, MYSQL
from util import print_table


def run():
    print("===== Test MySQL =====")
    conn = pymysql.connect(
        host=MYSQL["host"],
        port=MYSQL["port"],
        user=MYSQL["user"],
        password=MYSQL["password"],
        database=MYSQL["database"],
        connect_timeout=CONNECT_TIMEOUT_SEC,
    )
    try:
        with conn.cursor() as cur:
            cur.execute("SELECT * FROM Shippers")
            rows = cur.fetchall()
            columns = [d[0] for d in cur.description]
            print_table(columns, rows)
    finally:
        conn.close()


if __name__ == "__main__":
    run()
